cdap-security/src/main/java/co/cask/cdap/security/authorization/DefaultAuthorizationEnforcer.java

```diff
@@ -83,8 +83,25 @@ public void enforce(EntityId entity, Principal principal, Action action) throws
     }
 
     Set<? extends EntityId> difference = Sets.difference(entityIds, visibleEntities);
-    LOG.trace("Checking visibility of {} for principal {}.", difference, principal);
-    Set<? extends EntityId> moreVisibleEntities = authorizerInstantiator.get().isVisible(difference, principal);
+    LOG.trace("==> Checking visibility of {} for principal {}.", difference, principal);
+    // create new stopwatch instance every time enforce is called since the DefaultAuthorizationEnforcer is binded as
+    // singleton we don't want the stopwatch instance to get re-used across multiple calls.
+    StopWatch watch = new StopWatch();
+    watch.start();
+    Set<? extends EntityId> moreVisibleEntities;
+    try {
+      moreVisibleEntities = authorizerInstantiator.get().isVisible(difference, principal);
+    } finally {
+      watch.stop();
+      long timeTaken = watch.getTime();
+      String logLine = String.format("<== Checked visibility of %s for principal %s. Time spent in visibility check " +
```

Comment:
Can't we leverage LOG's string formatting capabilities? That'd be more efficient in most cases.

Reply:
What is the "<=="? Can you remove that?

Reply:
Time spent in check was {N} ms.

verb "was" is missing.